packages/@o3r/eslint-plugin/src/rules/typescript/matching-configuration-name/matching-configuration-name.ts

```diff
@@ -0,0 +1,76 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+import { createRule } from '../../utils';
+
+
+export default createRule({
+  name: 'matching-configuration-name',
+  meta: {
+    hasSuggestions: true,
+    fixable: 'code',
+    type: 'problem',
+    docs: {
+      description: 'Ensures that the configuration interface name matches the first parameter of `computeConfigurationName`',
+      recommended: 'error'
+    },
+    schema: [],
+    messages: {
+      error: '{{currentValue}} does not match: {{possibleValues}}',
+      suggestion: 'Replace {{currentValue}} by {{recommendedValue}}'
+    }
+  },
+  defaultOptions: [],
+  create: (context) => {
+    return {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      CallExpression: (node) => {
+        if (
+          node.callee.type !== TSESTree.AST_NODE_TYPES.Identifier
+          || node.callee.name !== 'computeConfigurationName'
+          || !node.arguments.length
+          || node.arguments[0].type !== TSESTree.AST_NODE_TYPES.Literal
+          || typeof node.arguments[0].value !== 'string'
+        ) {
+          return;
+        }
+
+        const sourceCode = context.getSourceCode();
+        const interfaceNames = sourceCode.ast.body
+          .filter((statement): statement is TSESTree.ExportNamedDeclaration & { declaration: TSESTree.TSInterfaceDeclaration } =>
+            statement.type === TSESTree.AST_NODE_TYPES.ExportNamedDeclaration
+            && statement.declaration?.type === TSESTree.AST_NODE_TYPES.TSInterfaceDeclaration
+            && (statement.declaration.extends || []).some((heritageClause) =>
+              heritageClause.expression.type === TSESTree.AST_NODE_TYPES.Identifier
+              && heritageClause.expression.name === 'Configuration'
```

Comment:
I would say that this is not working in case the `Configuration` interface has been aliased (but let's pretend we did not see it :D)